contributors/design-proposals/scheduling/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,186 @@
+
+Main Reviewers:
+
+Bobby [@bsalamat](https://github.com/bsalamat)
+Derek [@derekwaynecarr](https://github.com/derekwaynecarr)
+
+Dec 2017
+
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Changes in ResourceQuota](#changes-in-resourceQuota)
+    * [Changes in Admission Controller configuration](#changes-in-admission-controller-configuration)
+    * [Expected behavior of ResourceQuota admission controller and Quota system](#expected-behavior-of-resourcequota-admission-controller-and-resourcequota-system)
+      * [Backward Compatibility](#backward-compatibility)
+  * [Sample user story 1](#sample-user-story-1)
+  * [Sample user story 2](#sample-user-story-2)
+
+
+## Objective
+
+This feature is designed to make `ResourceQuota` become priority aware, several sub-tasks are included.
+
+1. Expand `Scopes` in `ResourceQuotaSpec` to represent priority class names and corresponding behavior.
+2. Incorporate corresponding behavior in quota checking process.
+3. Update the `ResourceQuota` admission controller to check priority class name and perform expected admission.
+
+### Non-Goals
+
+* Add priority in Pod spec (this is implemented separately in: [45610](https://github.com/kubernetes/kubernetes/pull/45610))
+
+## Background
+
+Since we already have [priority field in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610),
+Pods can now be classified into different priority classes. We would like to be able to create quota for various priority classes in order to manage cluster resources better and limit abuse scenarios.
+
+One approach to implement this is by adding priority class name field to `ResourceQuota` API definition. While this arbitrary field of API object will introduce inflexibility to potential change in future and also not adequate to express all semantics.
+
+Thus, we decide to reuse the existing `Scopes` of `ResourceQuotaSpec` to provide a richer semantics for quota to cooperate with priority classes.
+
+## Overview
+
+This design doc introduces how to define a group of priority class scopes for the quota to match with and explains how quota enforcement logic is changed to apply the quota to pods with the given priority classes.
+
+## Detailed Design
+
+### Changes in ResourceQuota
+
+ResourceQuotaSpec contains an array of filters, `Scopes`, that if mentioned, must match each object tracked by a ResourceQuota.
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+        ...
+
+        // A collection of filters that must match each object tracked by a quota.
+        // If not specified, the quota matches all objects.
+        // +optional
+        Scopes []ResourceQuotaScope
+}
+```
+Four new `ResourceQuotaScope` will be defined for matching pods based on priority class names.
```

Comment:
Please ensure during implementation that enforcement of these scopes and their presence on quota objects is gated by enabling the Priority feature gate.